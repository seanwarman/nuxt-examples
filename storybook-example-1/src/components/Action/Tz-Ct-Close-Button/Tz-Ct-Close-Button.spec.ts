/**
 * @vitest-environment jsdom
 */

import { mount, VueWrapper } from '@vue/test-utils';
import TzCtCloseButton from './Tz-Ct-Close-Button.vue';
import { expect, describe, beforeEach, it } from 'vitest';

let wrapper: VueWrapper;

describe('CT Close Button setup', () => {
  beforeEach(() => {
    wrapper = mount(TzCtCloseButton);
  });

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("emits a 'click' event when it's clicked", () => {
    wrapper.find('button').trigger('click');
    expect(wrapper.emitted()).toHaveProperty('click');
  });
});
