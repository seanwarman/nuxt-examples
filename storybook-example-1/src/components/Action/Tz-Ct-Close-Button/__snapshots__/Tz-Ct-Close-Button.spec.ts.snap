// Vitest Snapshot v1

exports[`CT Close Button setup > renders correctly 1`] = `
<button
  class="w-full items-center border-b-[1px] border-[#E3E5E3] bg-[#FBFBFB] p-4 text-xs font-extrabold text-dark sm:w-auto sm:border-b-0 sm:border-transparent sm:bg-transparent"
>
  <span
    class="underline"
  >
    CLOSE
  </span>
   COMPARISON TOOL 
</button>
`;
